kernel/memory_structs/src/lib.rs

```diff
@@ -24,6 +26,53 @@ use paste::paste;
 use derive_more::*;
 use range_inclusive::{RangeInclusive, RangeInclusiveIterator};
 
+/// Numeric constants associated with different page types.
+pub const PAGE_4KB_SIZE: usize = 1 << 12;
+pub const PAGE_2MB_SIZE: usize = (1 << 12) * 512;
+pub const PAGE_1GB_SIZE: usize = (1 << 12) * 512 * 512;
+
+/// Enum used to indicate the size of a page or frame.
+#[derive(Debug)]
+pub enum MemChunkSize {
+    Normal4K,
+    Huge2M,
+    Huge1G,
+}
+
+/// Trait used to group the size marker structs that paging-related types are parameterized with.
+pub trait PageSize: Ord + PartialOrd + Clone {
+    const SIZE: MemChunkSize;
+    const CONTAINED_4K: usize;
```

Comment:
nit: i find this name confusing, it'd probably be clearer to go with something like `SIZE_IN_4K_PAGES` or `NUM_4K_PAGES`